andela/bp-esa-backend
# #165083767 Get slack channels and save with partner records during automations

#### What does this PR do?
Ensure partner records with respective slack channels are saved during automation

#### Description of Task to be completed?
- redefine `createPartnerChannel` to `findOrCreatePartnerChannel`
- add `retrieve` enum key to slackAutomation type field
- redefine `updatePartnerStore` to `getPartnerFromStore`
- reimplement `findPartnerById` to ensure returned partner has slack details
- refactor slackOnboarding job for speed

#### How should this be manually tested?
1. On terminal, run command: `npm run start:dev`
2. Confirm that the partner details are saved for each client_id in the automation
3. Check slack automations to confirm that all off-boarding jobs are successful

#### Any background context you want to provide?
- When an off-boarding automation is to be carried, the automation fails because the partner is not found in database especially when the off-boarding placement was not automated by ESA, hence not having any slack channels to use in `kicking` the developer
- Even when partner detail is found, slack automation still fails because of absence of general slack channelId in the saved partner record

#### What are the relevant pivotal tracker stories?
[#165083767](https://www.pivotaltracker.com/story/show/165083767)

#### Postman Documentation
N/A
